src/tests/unit/tests/electron/views/automated-checks/automated-checks-view.test.tsx

```diff
@@ -30,30 +33,99 @@ describe('AutomatedChecksView', () => {
                 scanStoreData: {
                     status: ScanStatus.Scanning,
                 },
+                deviceStoreData: {},
             } as AutomatedChecksViewProps;
 
             const wrapped = shallow(<AutomatedChecksView {...props} />);
 
-            expect(wrapped.getElement()).toMatchSnapshot('automated checks view');
+            expect(wrapped.getElement()).toMatchSnapshot();
+        });
+
+        it('device disconnected', () => {
+            const props: AutomatedChecksViewProps = {
+                deps: {
+                    scanActionCreator: Mock.ofType(ScanActionCreator).object,
+                },
+                scanStoreData: {
+                    status: ScanStatus.Failed,
+                },
+                deviceStoreData: {
+                    connectedDevice: 'TEST DEVICE',
+                },
+            } as AutomatedChecksViewProps;
+
+            const wrapped = shallow(<AutomatedChecksView {...props} />);
+
+            expect(wrapped.getElement()).toMatchSnapshot();
         });
     });
 
     it('triggers scan when first mounted', () => {
-        const devicePort = 11111;
+        const port = 11111;
 
         const scanActionCreatorMock = Mock.ofType(ScanActionCreator);
-        scanActionCreatorMock.setup(creator => creator.scan(devicePort)).verifiable(Times.once());
+        scanActionCreatorMock.setup(creator => creator.scan(port)).verifiable(Times.once());
 
         const props: AutomatedChecksViewProps = {
             deps: {
                 scanActionCreator: scanActionCreatorMock.object,
             },
-            devicePort,
             scanStoreData: {},
+            deviceStoreData: {
+                port,
+            },
         } as AutomatedChecksViewProps;
 
         shallow(<AutomatedChecksView {...props} />);
 
         scanActionCreatorMock.verifyAll();
     });
+
+    describe('DeviceDisconnectedPopup event handlers', () => {
+        it('onRescanDevice', () => {
+            const port = 11111;
+
+            const scanActionCreatorMock = Mock.ofType(ScanActionCreator);
+
+            const props: AutomatedChecksViewProps = {
+                deps: {
+                    scanActionCreator: scanActionCreatorMock.object,
+                },
+                scanStoreData: {
+                    status: ScanStatus.Failed,
+                },
+                deviceStoreData: {
+                    port,
+                },
+            } as AutomatedChecksViewProps;
+
+            const wrapped = shallow(<AutomatedChecksView {...props} />);
+
+            wrapped.find(DeviceDisconnectedPopup).prop('onRescanDevice')();
+
+            scanActionCreatorMock.verify(creator => creator.scan(port), Times.exactly(2));
```

Comment:
This is confusing to me; why are there 2 scan messages rather than 1? If this is the expected behavior, can you leave a comment here explaining it?

Reply:
Oh, I see, there's a scan on first mount too. Maybe a more isolated test would be to clear the mock expectations between lines 102 and 104 and then just have this test verify the one call unique to the event handler?

Reply:
use the reset option, added a comment to explain why we do reset the mock! 
Thanks for the feedback.